software/base/src/main/java/brooklyn/entity/java/JavaSoftwareProcessSshDriver.java

```diff
@@ -336,7 +336,7 @@ protected int tryJavaInstall(String version, String command) {
         log.debug("Checking Java version at {}@{}", getEntity(), getLocation());
         // sed gets stdin like 'java version "1.7.0_45"'
         ProcessTaskWrapper<Integer> versionCommand = Entities.submit(getEntity(), SshTasks.newSshExecTaskFactory(
-                getLocation(), "java -version 2>&1 | grep \"java version\" | sed 's/.*\"\\(.*\\).*\"/\\1/'"));
+                getLocation(), "java -version 2>&1 | grep \" version\" | sed 's/.*\"\\(.*\\).*\"/\\1/'"));
```

Comment:
I've included a comment in the commit.
Here is what `java -version` returns on my side
```
openjdk version "1.8.0_45"
OpenJDK Runtime Environment (build 1.8.0_45-b14)
OpenJDK 64-Bit Server VM (build 25.45-b02, mixed mode)
```
However we could consider using `grep -E "(openjdk|java) version"`